students/Pirouz_N/lesson04/mailroom.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+"""
+Purpose: Mailroom Part 2 python certificate from UW
+Author: Pirouz Naghavi
+Date: 07/07/2020
+"""
+
+# imports
+import datetime
+
+donor_db = {}
+
+
+def write_db_to_file():
+    """"This function saves donors to the database in donor_db.txt located at current working sub directory data."""
+    with open('data/donor_db.txt', 'w') as output_file:
+        for db_key in donor_db:
+            output_file.write(('{}: ' + '{}, ' * (len(donor_db[db_key]) - 1) + '{}\n')
+                              .format(db_key, *donor_db[db_key]))
+            # Or using f string
+            # output_file.write(f'{db_key}: {str(donor_db[db_key])[1:len(str(donor_db[db_key])) - 1]}\n')
+
+
+def populate_db_from_file():
+    """"This function reads donors from database in donor_db.txt located at current working sub directory data."""
+    try:
+        with open('data/donor_db.txt', 'r') as input_file:
+            for line in input_file:
+                colon_sep_list = line.split(':')
+                donor_db[colon_sep_list[0]] = [float(item.strip()) for item in colon_sep_list[1].strip().split(',')]
+    except TypeError:
+        print('Database data has been corrupted. Please restore a previously functioning backup.')
+    except FileNotFoundError:
+        print('Database file is no available in the current working directory folder.')
+
+
+def exit_loop():
+    """"This function allows the user to exit main menu using global condition."""
+    global continue_loop
+    continue_loop = False
+
+
+def main_menu():
+    """"This function makes the menu portion of the program work."""
+    global continue_loop
+    continue_loop = True
+    option_dict = {'1': send_thank_you_after_donation, '2': create_report, '3': send_thank_you_to_all, '4': exit_loop}
+    while continue_loop:
+        option = input('Please select of the three options: Send a Thank You, Create a Report, send letters to all '
+                       'donors, or Quit.\n'
+                       'Type 1 and hit enter if you wish to Send a Thank You.\n'
+                       'Type 2 and hit enter if you wish to Create a Report.\n'
+                       'Type 3 and hit enter if you wish to send letters to all donors.\n'
+                       'Type 4 and hit enter if you wish to Quit.\n'
+                       'Please make your selection:\n')
+        try:
+            option_dict[option]()
+        except KeyError:
+            print('Please only type 1 or 2 or 3 or 4 and hit enter. Any other option will not be acceptable.')
```

Comment:
you can make this dynamic-> `print(f"Please choose from following options: {list(option_dict)}")`

Reply:
That's a great point. I completely missed that!
One small change I made just for aesthetics.  
print(f"Please choose from following options: {list(option_dict)}") will print:
Please choose from following options: ['1', '2', '3', '4']
I changed it a little to do what you where suggesting. Thank you. 
print(f"Please choose from following options: {str([int(item) for item in list(option_dict)])[1:-1]}")
Please always let me know if you can think of a better way to make my code look more professional :) 

Best Regards
Pirouz Naghavi